### PR TITLE
bip-0085: add Go reference implementation

### PR DESCRIPTION
**Pull request by satindergrewal**
Add [shurlinet/go-bip85](https://github.com/shurlinet/go-bip85) to the BIP85 reference implementations list.

- BIP85 v2.0.0 (includes DICE application)
- All 12 spec test vectors pass byte-for-byte
- 268 cross-implementation vectors generated from [bipsea](https://github.com/akarve/bipsea)
- Test patterns adopted from [ethankosakovsky/bip85](https://github.com/ethankosakovsky/bip85) (DRNG split reads, length edge cases)
- RSA supported via DRNG workaround (Go 1.24+ FIPS 140-3 constraint)
- MIT license
- [pkg.go.dev documentation](https://pkg.go.dev/github.com/shurlinet/go-bip85)

### CODE REVIEW

**Comment by satindergrewal** [comment]
looks like Typo check failed due to HTTP 404 downloading typos v1.46.1 binary - CI infrastructure issue, not a typo in the change.

File: bip-0085.mediawiki [hunk 1]
```diff
@@ -438,6 +438,7 @@ BIP32, BIP39
 * 1.3.0 Python 3.x library implementation: [https://github.com/akarve/bipsea]
 * 1.1.0 Python 2.x library implementation: [https://github.com/ethankosakovsky/bip85]
 * 1.0.0 JavaScript library implementation: [https://github.com/hoganri/bip85-js]
+* 2.0.0 Go library implementation: [https://github.com/shurlinet/go-bip85]
```
Comment (murchandamus) [inline bip-0085.mediawiki:441]: We do not allow collecting additional implementations in BIPs.

Via [BIP3](https://github.com/bitcoin/bips/blob/master/bip-0003.md#adoption-of-proposals):
> “It is not intended for BIPs to list additional implementations beyond the reference implementation: the BIPs repository is not a signpost where to find implementations.”

The [reasoning](https://github.com/bitcoin/bips/blob/master/bip-0003.md#user-content-fnref-OtherImplementations-2-c891adac936cbe49f5a9feee1ec8f46c) is given as follows:

> In the past, some BIPs had "Other Implementations" sections that caused frequent change requests to existing BIPs. This put a burden on the BIP authors and BIP Editors, and frequently led to lingering pull requests due to the corresponding BIPs’ authors no longer participating in the process. Many of these alternative implementations eventually became unmaintained or were low-quality to begin with. Therefore, "Other Implementations" sections are heavily discouraged.

**Comment by murchandamus** [comment]
I’m closing this PR, because it is not intended to link to a collection of numerous implementations in a BIP. cc: @akarve.
